File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
       let n=nums.length
       let pre=1
       let post=1
       let res=[]
       for(let i=0;i<n;i++)
       {
         res[i]=pre
         pre*=nums[i]
       }
       for(let i=n-1;i>=0;i--)
       {
         res[i]=res[i]*post
         post*=nums[i]
       }
       return res
}
}

File: Data Structures & Algorithms/majority-element/submission-0.go
func majorityElement(nums []int) int {
    var res= map[int]int {}

	for _,v:=range nums{
		res[v]++
	}

	for k,v:=range res{
		if v>len(nums)/2 {
			return k
		}
	}
	return 0
}
